server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -2490,7 +2490,11 @@ public NetworkResponse createNetworkResponse(ResponseView view, Network network)
         if (network.getVpcId() != null) {
             Vpc vpc = ApiDBUtils.findVpcById(network.getVpcId());
             if (vpc != null) {
-                response.setVpcId(vpc.getUuid());
+                try {
+                    _accountMgr.checkAccess(CallContext.current().getCallingAccount(), null, false, vpc);
+                    response.setVpcId(vpc.getUuid());
+                } catch (PermissionDeniedException e){
+                }
```

Comment:
this is strange, we check on access but there is no consequence?
at least we should log that something was denied/not setting or disclosing vpcid.

Reply:
@DaanHoogland , the reason to do this is to abstract the ID of the VPC from the account that does not have access to it; this way, the account would know the network belongs to a VPC, but would not be able to redirect to the VPC tab in the UI. Here follows an example, with VPCUser being accessible by the account and VPCRoot not being accessible:

![image](https://user-images.githubusercontent.com/49285692/216619535-a742f954-dcfd-4834-94f1-23687d2f09a2.png)


You are right, there should be a log there, I will add it.